fn/goroutine_manager_test.go

```diff
@@ -107,15 +108,50 @@ func TestGoroutineManagerStress(t *testing.T) {
 	// implementation, this test crashes under `-race`.
 	for i := 0; i < 100; i++ {
 		taskChan := make(chan struct{})
-		err := m.Go(func(ctx context.Context) {
+		ok := m.Go(func(ctx context.Context) {
 			close(taskChan)
 		})
 		// If goroutine was started, wait for its completion.
-		if err == nil {
+		if ok {
 			<-taskChan
 		}
 	}
 
 	// Wait for Stop to complete.
 	<-stopChan
 }
+
+// TestGoroutineManagerStopsStress launches many Stop() calls in parallel with a
+// task exiting. It attempts to catch a race condition between wg.Done() and
+// wg.Wait() calls. According to documentation of wg.Wait() this is acceptable,
+// therefore this test passes even with -race.
```

Comment:
Given the implementation of WaitGroups I think we don't need this Stresstest anymore or ? There is no problem in calling `wg.Done()`and potentially reaching a counter of 0. Only adding new Goroutines are a problem and cause potential race conditions.

Reply:
I would keep it as a safeguard in case we change the implementation of GoroutineManager in the future, and the new implementation is not resilient to this scenario.